Update plan cache doc (#7889)
pingcap/docs: sql-prepare-plan-cache.md

- The query contains `?` in the definition of the `Window Frame` window function, such as `(partition by year order by sale rows ? preceding)`. If `?` appears elsewhere in the window function, the query is cached.
- The query contains parameters for comparing `int` and `string`, such as `c_int >= ?` or `c_int in (?, ?)`, in which `?` indicates the string type, such as `set @x='123'`. To ensure that the query result is compatible with MySQL, parameters need to be adjusted in each query, so such queries are not cached.
- The plan attempts to access `TiFlash`.

> - In most cases, the plan that contains `TableDual` will not be cached, unless the current `Prepare` statement does not have parameters.

Author: ```suggestion
- In most cases, the plan that contains `TableDual` is cached, unless the current `Prepare` statement does not have parameters.
```
